web_ui/frontend/components/layout/Sidebar/UserMenu.tsx

```diff
@@ -0,0 +1,109 @@
+'use client'
+
+import useSWR from "swr";
+import {useRouter} from "next/navigation";
+import React, {useContext, useState} from "react";
+import {IconButton, Menu, MenuItem} from "@mui/material";
+import {Login} from "@mui/icons-material";
+import StatusSnackBar from "@/components/StatusSnackBar";
+import {RandomUserIcon} from "@/components/layout/Sidebar/RandomUserIcon";
+import {getUser} from "@/helpers/login";
+
+const UserMenu = () => {
+
+    const userMenuRef = React.useRef(null);
+
+    const {data: user, isLoading, error: fetchError, mutate} = useSWR(
+        "getUser",
+        getUser,
+        {
+            refreshInterval: 1000 * 60,
+            fallbackData: {authenticated: false}
+        }
+    )
+
+    const router = useRouter()
+
+    const [menuOpen, setMenuOpen] = useState(false)
+    const [error, setError] = useState<string | undefined>(undefined)
+
+    const handleLogout = async (e: React.MouseEvent<HTMLElement>) => {
+        try {
+            let response = await fetch("/api/v1.0/auth/logout", {
+                method: "POST",
+                headers: {
+                    "Content-Type": "application/json"
+                }
+            })
+
+            if(response.ok) {
+                await mutate(getUser)
+            } else {
+                try {
+                    let data = await response.json()
+                    if (data?.error) {
+                        setError(response.status + ": " + data['error'])
+                    } else {
+                        setError("Server error with status code " + response.status)
+                    }
+                } catch {
+                    setError("Server error with status code " + response.status)
+                }
+            }
+        } catch {
+            setError("Could not connect to server")
+        }
+    }
+
+    if(!user.authenticated){
+        return (
+            <IconButton
+                id={"user-menu-button"}
+                ref={userMenuRef}
+                sx={{
+                    bgcolor: "#767adb4a"
+                }}
+                onClick={() => router.push("/login" + "?returnURL=" + window.location.pathname)}
+            >
+                <Login/>
+            </IconButton>
+        )
+    }
+
+    return (
+        <>
+            <IconButton
+                id={"user-menu-button"}
+                ref={userMenuRef}
+                sx={{
+                    bgcolor: "#4dba5a3b"
+                }}
+                onClick={() => setMenuOpen(!menuOpen)}
+            >
+                <RandomUserIcon/>
```

Comment:
Can we have a hover-over tooltip here as well? We can say "Admin User" | "General User" or just "Profile"

Reply:
Also, since we display the user status anyway, do we want to distinguish between admin user and general user by the UserIcon (besides having "admin" as the menu item below)?